src/parse.cpp

```diff
@@ -374,6 +374,7 @@ expression* parser::parse_primary_expression(precedence prec) {
 
   // class {}
   case token_type::kw_class: {
+    class_guard g(this, std::exchange(this->in_class_, true));
     expression* class_expression = this->parse_class_expression();
```

Comment:
Could we put the `class_guard` inside `parse_and_visit_class_body` instead to avoid repeating the `class_guard` code in the callers?